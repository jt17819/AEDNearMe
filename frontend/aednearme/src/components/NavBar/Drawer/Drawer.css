@import url('https://fonts.googleapis.com/css2?family=Enriqueta&display=swap');

body {
  margin: 0 auto;
  background-color: #ba181b !important;
  color: #fff !important;
  padding: 20px, 20px, 20px, 20px !important;
  font-family: 'Enriqueta', serif !important;
}