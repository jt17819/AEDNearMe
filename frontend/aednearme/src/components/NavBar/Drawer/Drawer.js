import React, { useState } from "react";
import {
  Divider,
  Drawer,
  IconButton,
  List,
  ListItem,
  ListItemText,
  makeStyles
} from "@material-ui/core";
import { Link } from "react-router-dom";
import MenuIcon from "@material-ui/icons/Menu";
import Menu from '@mui/material/Menu';
import { MenuItem } from "@mui/material";
import MobileMenu from "../../DropdownMenu/MobileMenu";

const useStyles = makeStyles(() => ({
  drawer: {
    background: 'linear-gradient(190deg, #ba181b, #000)',
    paddingBottom: '100px',
  },
  logoBox: {
    display: "flex",
    justifyContent: "end",
    alignItems: "center",
  },
  link: {
    textDecoration: "none",
    color: "#fff",
    // fontSize: "20px",
    // fontWeight: "bold",
  },
  icon: {
    color: "#fff",
    // 64f227 bright green color
  },
  logo: {
    flexGrow: "1",
    cursor: "pointer",
    textDecoration: "none",
    color: "#fff",
    fontSize: "20px",
    fontWeight: "bolder",
    lineHeight: "0px"
  },

}));

function DrawerComponent() {
  const classes = useStyles();
  const [openDrawer, setOpenDrawer] = useState(false);

  const handleLogout = () => {
    sessionStorage.clear()
    alert('You have been logged out')
    window.location.reload()
  }

  return (
    <>
      <Drawer 
        open={openDrawer}
        onClose={() => setOpenDrawer(false)}
        classes={{ paper: classes.drawer }}
      >
        <List
          style={{ background: '#ba181b' }}
        >
          <ListItem onClick={() => setOpenDrawer(false)}>
            <ListItemText>
              <Link to="/" className={classes.link}>Home</Link>
            </ListItemText>
          </ListItem>

          <Divider />

          <MobileMenu title="AED" 
            menuItems={aedArray} 
            handleClose={() => setOpenDrawer(false)} 
          />

          <Divider />

          <MobileMenu title="CPR" 
            menuItems={cprArray} 
            handleClose={() => setOpenDrawer(false)} 
          />

          <Divider />

          <MobileMenu title="Choking" 
            menuItems={chokingArray} 
            handleClose={() => setOpenDrawer(false)} 
          />

          <Divider />

          <MobileMenu 
            title="Education Hub" 
            menuItems={educationArray} 
            handleClose={() => setOpenDrawer(false)} 
          />

          <Divider />

          <ListItem onClick={() => setOpenDrawer(false)}>
            <ListItemText>
            {
                sessionStorage.getItem('accessToken') ?
                <Link to="/" onClick={handleLogout} className={classes.link}>Logout</Link> :
                <Link to="/login" className={classes.link}>Login</Link>
              }
            </ListItemText>
          </ListItem>

        </List>

      </Drawer>

      <div className={classes.logoBox}>

        <IconButton
          sx={{ fontSize: 80 }}
          onClick={() => setOpenDrawer(!openDrawer)} className={classes.icon}>
          <MenuIcon />
        </IconButton>

        <span >
          <Link className={classes.logo} to="/">AED <br /> nearME </Link>
        </span>
      </div>

    </>
  );
}
export default DrawerComponent;



const aedArray = [
  {
    title: "How to use",
    path: '/aedhowto'
  },
  {
    title: "Add NEW location",
    path: '/upload'
  }
]

const cprArray = [
  {
    title: "How to give",
    path: "/cprhowto"
  }
]

const chokingArray = [
  {
    title: "How to help",
    path: "/choking"
  }
]

const educationArray = [
  {
    title: "QUIZ",
    path: "/quiz"
  },
  {
    title: "AED: How to use",
    path: "/aedhowto"
  },
  {
    title: "CPR: How to give",
    path: "/cprhowto"
  },
  {
    title: "Choking: How to help",
    path: "/choking"
  }
]

