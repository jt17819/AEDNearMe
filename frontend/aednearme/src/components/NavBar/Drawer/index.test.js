import { default as DrawerComponent } from "./Drawer";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { MemoryRouter } from "react-router-dom";
import {BrowserRouter as Router} from "react-router-dom"


describe("Drawer", () => {
   let mockFunction;
   beforeEach(() => {
     mockFunction = jest.fn();
     // eslint-disable-next-line testing-library/no-render-in-setup
     render( <DrawerComponent />, { wrapper: MemoryRouter } );
   });

   test("It renders the AEDnearME link", () => {
        let link = screen.getByRole("link")
        expect(link).toBeInTheDocument();
     });

    //  test("it renders", () => {
    //    const navigation = screen.getByRole("nav");
    //    expect(navigation.textContent).toMatch("");
    //  });

    it('should take a snapshot', () => {
        // render(<DrawerComponent />, { wrapper: MemoryRouter })
        const { asFragment } = render(<DrawerComponent />, { wrapper: MemoryRouter })
        expect(asFragment(<DrawerComponent />)).toMatchSnapshot()
      })

   });
