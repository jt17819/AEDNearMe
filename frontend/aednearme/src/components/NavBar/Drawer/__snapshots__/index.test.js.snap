// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-logoBox-7"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root makeStyles-icon-9"
      sx="[object Object]"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <span>
      <a
        class="makeStyles-logo-10"
        href="/"
      >
        AED 
        <br />
         nearME 
      </a>
    </span>
  </div>
</DocumentFragment>
`;
